rewards/utils/rewards_utils.py

```diff
@@ -78,6 +79,32 @@ def distribute_rewards_to_snapshot(
     return combine_rewards([rewards] + custom_rewards_list, 0)
 
 
+def distribute_rewards_from_total_snapshot(
+        amount: int, snapshot: Snapshot, token: str,
+        custom_rewards: Optional[Dict[str, Callable]] = None,
+):
+    if not custom_rewards:
+        custom_rewards = {}
+    rewards = RewardsList()
+    custom_rewards_list = []
+    total = snapshot.total_balance()
+    # TODO: Think about refactoring this and splitting it into two separate funcs:
+    # TODO: one for normal rewards another for custom rewards
+    for addr, balance in snapshot:
+        if addr in custom_rewards:
+            custom_rewards_calc = custom_rewards[addr]
+            console.log(token, amount, snapshot.token)
+            custom_rewards_list.append(
+                custom_rewards_calc(amount, token, snapshot.token)
+            )
+        else:
+            rewards_percentage = Decimal(balance) / total if not total == 0 else 0
+            reward_amount = Decimal(amount) * rewards_percentage
+            assert reward_amount >= 0
+            rewards.increase_user_rewards(addr, token, reward_amount)
+    return combine_rewards([rewards] + custom_rewards_list, ZERO_CYCLE)
```

Comment:
Technically setting the cycle value here to zero has no effect on the rewards due the fact that this rewards list gets merged into the tree data. It might be worth to have this function default to zero so that we do not need pass this in

Reply:
I suggest that if that is not critical, let's leave that for the future refactorings. This PR is already getting scary big and we don't want to anything else new